### PR TITLE
New FromRequest[Parts] trait cleanup

### PR DESCRIPTION
**Pull request by jplatte**
Who would have thought that it's so easy? 😅 (the tuple stuff)

### CODE REVIEW

File: axum-core/src/extract/tuple.rs [hunk 1]
```diff
@@ -53,10 +48,10 @@ macro_rules! impl_from_request {
         // implementation of `FromRequest<S, B, Mut>` for `T: FromRequestParts<S>`.
         #[async_trait]
         #[allow(non_snake_case, unused_mut, unused_variables)]
-        impl<S, B, $($ty,)* $last> FromRequest<S, B, private::TupleOnce> for ($($ty,)* $last,)
+        impl<S, B, $($ty,)* $last> FromRequest<S, B> for ($($ty,)* $last,)
         where
             $( $ty: FromRequestParts<S> + Send, )*
-            $last: FromRequest<S, B, super::private::Once> + Send,
+            $last: FromRequest<S, B> + Send,
```
Comment (jplatte) [inline axum-core/src/extract/tuple.rs:54]: If we want to write some docs about how to write generic extractors that wrap other extractors (that might or might not consume the request), this seems like pretty straight-forward advice: Implement both `FromRequestParts` and `FromRequest`, without any generic-ness over `M` in the latter case.

**Comment by jplatte** [comment]
One more thought: Maybe we should rename `Mut` to `ViaParts` now that the traits are no longer `FromRequestOnce`, `FromRequestMut`? Not sure about the `Once` name.

**Review by davidpdrsn — APPROVED** [review]
Haha awesome!

File: axum-core/src/extract/tuple.rs (hunk 1 above)
Comment (davidpdrsn) [reply axum-core/src/extract/tuple.rs:54]: Yes! Its on the todo list [here](https://github.com/tokio-rs/axum/pull/1272)

**Comment by davidpdrsn** [comment]
> One more thought: Maybe we should rename Mut to ViaParts now that the traits are no longer FromRequestOnce, FromRequestMut? Not sure about the Once name.

@jplatte `ViaParts` and `ViaRequest` maybe?

**Pull request merged**